javaparser-core/src/main/javacc/java.jj

```diff
@@ -2227,6 +2229,21 @@ SwitchStmt SwitchStatement():
   { return new SwitchStmt(range(begin, token()), selector, entries); }
 }
 
+SwitchExpr SwitchExpression():
+{
+	Expression selector;
+	SwitchEntryStmt entry;
+	NodeList<SwitchEntryStmt> entries = emptyList();
+	JavaToken begin;
+}
+{
+  "switch" {begin=token();} "(" selector = Expression() ")" "{"
+    ( entry = SwitchEntry() { entries = add(entries, entry); } )*
+  "}"
+
+  { return new SwitchExpr(range(begin, token()), selector, entries); }
```

Comment:
This is and will probably stay the same as SwitchStatement(), not sure if javacc has a nice way of combining the two.